Cargo.toml

```diff
@@ -120,11 +120,9 @@ jj-lib = { path = "lib", version = "0.16.0" }
 jj-lib-proc-macros = { path = "lib/proc-macros", version = "0.16.0" }
 testutils = { path = "lib/testutils" }
 
-# Insta suggests compiling these packages in opt mode for faster testing.
-# See https://docs.rs/insta/latest/insta/#optional-faster-runs.
-[profile.dev.package]
-insta.opt-level = 3
-similar.opt-level = 3
+[profile.dev.package."*"]
+codegen-units = 1
```

Comment:
I think this is what makes it so much slower. I agree with Yuya that the size of the binary is not important to optimize for.

I'll happily approve a PR that just sets `opt-level = 3` for all dependencies and nothing else (but it's a significant change, so you might want to wait to hear from others too).

Reply:
FWIW, `codegen-units=1` has nearly no impact at all on compile times (or at least the fluxuations are minor), and also has almost no (repeatably measurable) impact on runtime for me, either. I included it for a more pedestrian reason: so that the dependencies in the development build more closely match the dependencies in the release build, where we do set that option. But I can drop it too, if we'd like.

Reply:
And also, I want to be clear it is *not* the size of the final binaries that I am worried about with the `line-tables-only` change. It is *the size of every intermediate artifact ever compiled* &mdash; by rust-analyzer, by running nextest, from dependabot updates, etc. My development machine for `jj` has 1TB of space, but it's absurd that Cargo can eat almost +100GiB every two weeks for a single project. This will continue to get worse as we pile in more crates (but that's another discussion)

There are other factors here, for example my machine uses mold for linking, but I suspect that this change would have a positive impact on non-mold users (where there is no parallelism, and debuginfo tends to have a disproportionate affect on link time). I also suspect that on an actual HDD where write times are 100x slower and random reads are slow and sequential will probably see improvements, too. But those are just speculation; I haven't tested them, though I'm happy to.

Reply:
On my machine, `codegen-units=1` on dev deps increases the build time from about 50s to about 60s. Since dependencies are rarely recompiled, I'm fine with either.